rfcs/text/0000-multi-data-collection-dataset-processing.md

```diff
@@ -0,0 +1,115 @@
+
+In the future, there may be a need to run analysis processes where input sets span multiple projects. A mechanism to specify processing data collections that are not restricted to a single project or submissions will be required to implement this type of analysis.
+
+#### Algorithmic complexity and performance
+Producing combined metadata or data from multiple assay bundles using the bundle event system results in O(N^2) algorithmic complexity, where N is the number of data bundles in some grouping.  As each bundle event arrives, it must be combined with accumulated metadata from previous bundles.  Additionally, there is a race condition on reading and writing the accumulated the results that must be handled.
+
+One approach to addressing this is to just queue all data bundle events, however, this still requires knowledge of completion so that process can be triggered.
+
+This impacts the data browser creating project metadata TSVs and normalized JSON files, and the matrix service creating pre-built project matrices.
+
+#### Data consistency
+Consistency and completeness across a data set is an important attribute of an atlas.  An incomplete data set may result in misinterpretation of data or missed discoveries.  While a data set can be updated in the future, there is a need that to know when the current version of a data set is fully available.  This is refered to as *version-completeness*. Currently, the DCP has no way to indicate version-completeness beyond a bundle. The data browser and a consumer API and  don't have the information indicate a project submission is incomplete.  This information is only available within ingest, with no way to communicate submission status to other components.
+
+#### Data set quality control
+An approach to quality control (QC) is to run a partial or full analysis on a subset of the data before doing a full analysis.  This requires defining an analyzable subset of the data to pass on to the processing pipelines. A QC data subset must meet the criteria that allow the QC analysis to run.
+
+#### Data releases
+A data release is a grouping of bundles meeting some criteria of consistency and quality that is considered of value to the users to be addressed as a whole.  Data releases require grouping of specific versions of data in ways that can not be defined before release time.  There may be data quality and project reasons for sub-setting data in a release a given manner.
+
+The DSS currently support functionality called *collections*, which some have suggested can be used to support releases.  However, documentation of the semantics and functional restrictions can not be found, so it is unclear if this mechanism is useful.
```

Comment:
The swagger doc is more helpful than readthedocs but not really sufficient.  I have no idea what a metadata fragment is or why one would include it.  The limit on 1000 items per put/patch makes it very undesirable for something that can contain millions of items.

There is work to improve the DSS documentation, so hopefully, this will work itself out.